source/server/spongineer/debugging.rst

```diff
@@ -119,3 +119,102 @@ A ``NoClassDefFoundError`` occurs when the plugin tries to access a class that i
+    /*     net.minecraft.world.World.func_72866_a(World.java:4154)                                                                */
+    /*     net.minecraft.world.WorldServer.func_72866_a(WorldServer.java:832)                                                     */
+    /*     net.minecraft.world.World.func_72870_g(World.java:1952)                                                                */
+    /*     net.minecraft.world.World.func_72939_s(World.java:6596)                                                                */
+    /*     net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:2300)                                                    */
+    /*     net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:767)                                            */
+    /*     net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396)                                  */
+    /*     net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)                                            */
+    /*     net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)                                                     */
+    /*     java.lang.Thread.run(Unknown Source)                                                                                   */
+    /*  Phases Remaining:                                                                                                         */
+    /*                                                                                                                            */
+    /* Minecraft : 1.12.2                                                                                                         */
+    /* SpongeAPI : 7.1.0-SNAPSHOT-7105dfc                                                                                         */
+    /* SpongeForge : 1.12.2-2705-7.1.0-BETA-3361                                                                                  */
+    /* Minecraft Forge : 14.23.4.2705                                                                                             */
+    /******************************************************************************************************************************/
+
+This stacktrace contains the most important version numbers, as well some information about the phase the server was in.
+In this case a ``NullPointerException`` has been thrown during ``EntityTickPhase``. At this point its important to check
+which plugins are involved with the crash. This requires some research as you have to match the package names with mod
+names (Also check the ``Caused by`` blocks).
+
+* ``java`` classes can be ignored during the error search.
+* ``net.minecraft`` is the vanilla Minecraft code. If only these elements are present, its either a Minecraft bug or a
+  really nasty one (I explain those later).
```

Comment:
its -> it's
We try to avoid first-person wherever possible, so can you change (I explain ...) to (these are explained below).
-- As an afterthought: do we have a calibrated scale of *Nasty* ?

Reply:
No, there is no such scale. However the nasty scale is tightly coupled with the required effort to find/analyze/understand/fix the issue.

Reply:
Maybe "hard to find"?

Reply:
Your choices of {`nasty` < `really nasty`} are sufficient.